syntax = "proto3";

option go_package = "envoy";

package pb.cilium;

message KeyValue {
  string key = 1;
  string value = 2;
}

enum Protocol {
  HTTP10 = 0;
  HTTP11 = 1;
  HTTP2 = 2;
}

enum EntryType {
  Request = 0;
  Response = 1;
  Denied = 2;
}

message HttpLogEntry {
  // The time that Cilium filter captured this log entry,
  // in, nanoseconds since 1/1/1970.
  uint64 timestamp = 1;

  Protocol http_protocol = 2;

  EntryType entry_type = 3;

  // Cilium Redirect resource name
  string cilium_resource_name = 4;

  // Cilium rule reference
  string cilium_rule_ref = 5;
  
  // Cilium security ID of the source
  uint32 source_security_id = 6;

  // These fields record the original source and destination addresses,
  // stored in ipv4:port or [ipv6]:port format.
  string source_address = 7;
  string destination_address = 8;

  // The HTTP response code
  uint32 response_code = 9;

  repeated KeyValue headers = 10;
}
